### DISCUSSION TITLE
[Feature Request] Gantt charts

### DISCUSSION THREAD

**Discussion by AuroraV** [apache/superset#23503] [Ideas]
Make sure these boxes are checked before submitting your issue - thank you!

- [x] I have checked the issue tracker for the same issue and I haven't found one similar
Is there any way to do this currently with the existing visualisations, or any plans to implement this? (a search did not bring up anything).

### Motivation
Really need project management charts, e.g. gantt.

Looks like there is an existing Gantt chart library for D3 if that is of any interest: https://github.com/dk8996/Gantt-Chart

### Proposed Change
Issue https://github.com/apache/superset/issues/4035 , https://github.com/apache/superset/issues/8792 already got closed. PR https://github.com/apache/superset/pull/5174 seems to be complete and also got closed. Unfortunately, not very technical in terms of Superset and can't comment on it.

**Comment by Stefan781** [comment]
+1

**Comment by Codes-Lab** [comment]
+1

**Comment by vladradishevsky** [comment]
+1

**Comment by abdulrk401** [comment]
Your best bet is to implement a custom chart plugin: https://preset.io/blog/building-custom-viz-plugins-in-superset-v2/

Since echarts is mostly supported, the best way to get started is by looking at current examples. Here's a few I could find:
- https://echarts.apache.org/examples/en/editor.html?c=custom-gantt-flight
- https://echarts.apache.org/examples/en/editor.html?c=custom-profile

It'll take some work but definitely worth it.

**Reply by Stefan781** [reply]
I looked at those particular ECharts some time ago and arrived at the same conclusion. Just a note (as far as my understanding goes): the two linked charts are actually custom charts so the first step would be to understand how those work and how can they be generalized. In essence, you would be building a Gantt chart extension to ECharts that you will then port to Superset. So it will be indeed some work.

**Reply by abdulrk401** [reply]
Yes, you are correct.

**Comment by stasDomb** [comment]
+1

**Comment by fliplafe** [comment]
+1

**Comment by tanlda** [comment]
+1

**Comment by AmeliaPessoa** [comment]
+1

**Comment by sfirke** [comment]
FYI there has been a SIP for this: https://github.com/apache/superset/issues/33349 and there's now a pull request open to merge the feature: https://github.com/apache/superset/pull/33716

**Comment by sunTimer** [comment]
+10086

**Comment by jcbrugiroux** [comment]
+1

**Comment by sfirke** [comment]
Gantt charts were added to Superset in https://github.com/apache/superset/pull/33716, that feature is available in versions 6.0.0 and later.
